src/services/TextReader.hh

```diff
@@ -2,7 +2,7 @@
 #define RA_SERVICES_TEXTREADER
 #pragma once
 
-#include <string>
+#include <xstring>
```

Comment:
We just need the using for the string. forward declaring `std::basic_string `is kind of difficult, so this should be good enough.